File: surf-api-bukkit/surf-api-bukkit-server/src/main/java/dev/slne/surf/surfapi/bukkit/server/reflection/ServerStatsCounterProxy.java
package dev.slne.surf.surfapi.bukkit.server.reflection;

import net.minecraft.stats.ServerStatsCounter;
import xyz.jpenilla.reflectionremapper.proxy.annotation.MethodName;
import xyz.jpenilla.reflectionremapper.proxy.annotation.Proxies;

@Proxies(ServerStatsCounter.class)
public interface ServerStatsCounterProxy {

  @MethodName("toJson")
  String toJson(ServerStatsCounter statsCounter);
}

File: surf-api-paper/surf-api-paper-server/src/main/kotlin/dev/slne/surf/api/paper/server/reflection/ServerStatsCounterProxy.kt
package dev.slne.surf.api.paper.server.reflection

import com.google.gson.Gson
import com.google.gson.JsonElement
import net.minecraft.stats.ServerStatsCounter
import xyz.jpenilla.reflectionremapper.proxy.annotation.FieldGetter
import xyz.jpenilla.reflectionremapper.proxy.annotation.MethodName
import xyz.jpenilla.reflectionremapper.proxy.annotation.Proxies
import xyz.jpenilla.reflectionremapper.proxy.annotation.Static

@Proxies(ServerStatsCounter::class)
interface ServerStatsCounterProxy {

    @MethodName("toJson")
    fun toJson(statsCounter: ServerStatsCounter): JsonElement

    @FieldGetter("GSON")
    @Static
    fun getGson(): Gson
}
